prow/plugins/trigger/trigger.go

```diff
@@ -155,10 +155,14 @@ func handlePush(pc plugins.Agent, pe github.PushEvent) error {
 func TrustedUser(ghc trustedUserClient, onlyOrgMembers bool, trustedOrg, user, org, repo string) (bool, error) {
```

Comment:
This can be called multiple times per handled event right? This seems like it could be prohibitively expensive unless ghProxy is enabled which we can't yet assume. Instead of working around this can we first see if GH will make their API properly generate Etags for the isCollaborator call?

Reply:
So this would increase token use for repos where this call will scale with the number of pages it needs to read, but I don't see an alternative right now.